### PR TITLE
Fix failing wpa_supplicant test

### PR DESCRIPTION
**Pull request by grisu48**
Attempt to fix the failing wpa_supplicant test on Tumbleweed by introducing two changes:

* Don't fail wpa_supplicant if an error happens during the cleanup routine (e.g. killing an already terminated process)
* The test now passes a cookie to openQA if and only if the test passes. We ignore the return code of the script and await the cookie to mark the test as successful.

- Related ticket: https://progress.opensuse.org/issues/106676
- Verification run: https://duck-norris.qam.suse.de/tests/8478#step/wpa_supplicant/38 (wpa_supplicant passes, failing in other steps)

### CODE REVIEW

File: data/wpa_supplicant/wpa_supplicant_test.sh
Comment (asmorodskyi) [inline data/wpa_supplicant/wpa_supplicant_test.sh:119]: why we are ok if wicked has failed state here ?

Comment (grisu48) [reply data/wpa_supplicant/wpa_supplicant_test.sh:119]: Good catch, I need to set `pipefail` as well. The reason for this is that otherwise the script blocks when executed outside of openQA

Comment (grisu48) [reply data/wpa_supplicant/wpa_supplicant_test.sh:120]: I've added `set -o pipefail`, should be ok now. Please mark as resolved if you agree

**Comment by asmorodskyi** [comment]
but I would wait for @cfconrad  before merge

File: data/wpa_supplicant/wpa_supplicant_test.sh
Comment (cfconrad) [inline data/wpa_supplicant/wpa_supplicant_test.sh:125]: I think we should judge from wicked perspective.

```suggestion
    wicked --log-level debug --debug all ifup wlan1 --timeout 90 2>&1 | tee wicked.log
```

**Review by cfconrad — COMMENTED** [review]
General agree, I only would change the `wicked ifup` part as suggested, wdyt?

File: data/wpa_supplicant/wpa_supplicant_test.sh
Comment (grisu48) [reply data/wpa_supplicant/wpa_supplicant_test.sh:125]: Please see the comment just above this command:

    # wicked ifup might terminate with setup-in-progress (rc=162), so check the link manually

This is one of the reasons why this test occasionally fails. AFAICS `wicked` terminates prematurely with rc=162, but the link is setup correctly. In this state the test fails because we have `-e` set.

Comment (cfconrad) [reply data/wpa_supplicant/wpa_supplicant_test.sh:125]: yes, thats why I increased from 60 to 90 the timeout as you also do this `60 + sleep 30`. 
And I actually wonder why this failing. Even 60 seconds should be enough in all circumstances here.

Comment (grisu48) [reply data/wpa_supplicant/wpa_supplicant_test.sh:125]: In my test runs it never waited for even the 60 seconds but terminated after just a couple of them, so increasing the timeout alone is not sufficient.

Comment (cfconrad) [reply data/wpa_supplicant/wpa_supplicant_test.sh:125]: oh, that is something I didn't expect.  I need to check it then lets first go with your change to unblock the tests. 
Wicked wlan is tested in wicked-testsuite https://github.com/os-autoinst/os-autoinst-distri-opensuse/tree/master/tests/wicked/wlan/sut

**Review by cfconrad — APPROVED** [review]

File: data/wpa_supplicant/wpa_supplicant_test.sh
Comment (cfconrad) [reply data/wpa_supplicant/wpa_supplicant_test.sh:125]: I was to fast... now I realized your words :)

```bash
wicked --debug ifup wlan1 --timeout 60
Bad debug facility "ifup"
wicked [options] <command> ...
```
This is what I get with your command. I guess it should be `--debug all` ...

**Comment by ostruzina** [comment]
Hi Felix, we're seeing failures in maintenance jobs after this change since today: https://openqa.suse.de/tests/8470105#step/wpa_supplicant/29

**Pull request merged**
